includes/checkout/functions.php

```diff
@@ -414,3 +414,184 @@ function edd_enforced_ssl_asset_filter( $content ) {
 
 	return $content;
 }
+
+/**
+ * Given a number and algorithem, determine if we have a valid credit card format
+ *
+ * @since  2.4
+ * @param  integer $number The Credit Card Number to validate
+ * @param  string  $method The algorithem to use, by default our build in Luhn method
+ * @return bool            If the card number provided matches a specific format of a valid card
+ */
+function edd_validate_card_number_format( $number = 0, $method = 'luhn' ) {
+
+	$number = trim( $number );
+	if ( empty( $number ) ) {
+		return false;
+	}
+
+	if ( ! is_numeric( $number ) ) {
+		return false;
+	}
+
+	$is_valid_format = false;
+
+	// First check if it passes with the passed method, Luhn by default
+	$is_valid_format = edd_validate_card_number_format_luhn( $number );
+
+	// Run additional checks before we start the regexing and looping by type
+	$is_valid_format = apply_filters( 'edd_valiate_card_format_pre_type', $is_valid_format, $number );
+
+	if ( true === $is_valid_format ) {
+		// We've passed our methdd check, onto card specific checks
+		$card_type       = edd_detect_cc_type( $number );
+		$is_valid_format = ! empty( $card_type ) ? true : false;
+	}
+
+	return apply_filters( 'edd_cc_is_valid_format', $is_valid_format, $number, $method );
+}
+
+/**
+ * Validate credit card number based on the luhn algorithm
+ *
+ * @since  2.4
+ * @param string $number
+ * @return bool
+ */
+function edd_validate_card_number_format_luhn( $number ) {
+
+	// Strip any non-digits (useful for credit card numbers with spaces and hyphens)
+	$number = preg_replace( '/\D/', '', $number );
+
+	// Set the string length and parity
+	$number_length = strlen( $number );
```

Comment:
Can we align these please? Change `number_length` to just `length` for better alignment.